addons/wowl/static/tests/components/dropdown_tests.ts

```diff
@@ -0,0 +1,475 @@
+      </Dropdown>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler");
+  await click(parent.el!, "ul.o_dropdown_menu li");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("payload received on item selection", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown, DropdownItem };
+    static template = owl.tags.xml`
+      <Dropdown t-on-dropdown-item-selected="onItemSelected">
+        <t t-set-slot="menu">
+          <DropdownItem payload="{ answer: 42 }"/>
+        </t>
+      </Dropdown>
+    `;
+    onItemSelected(ev: OwlEvent<any>) {
+      assert.deepEqual(ev.detail.payload, { answer: 42 });
+    }
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler");
+  await click(parent.el!, "ul.o_dropdown_menu li");
+});
+
+QUnit.test("multi-level dropdown: can be rendered and toggled", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`
+      <Dropdown>
+        <t t-set-slot="menu">
+          <Dropdown>
+            <t t-set-slot="menu">
+              <Dropdown/>
+            </t>
+          </Dropdown>
+        </t>
+      </Dropdown>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 3);
+});
+
+QUnit.test("multi-level dropdown: initial open state can be true", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`
+      <Dropdown open="true">
+        <t t-set-slot="menu">
+          <Dropdown open="true">
+            <t t-set-slot="menu">
+              <Dropdown open="true"/>
+            </t>
+          </Dropdown>
+        </t>
+      </Dropdown>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  assert.containsN(parent.el!, "ul.o_dropdown_menu", 3);
+});
+
+QUnit.test("multi-level dropdown: close on outside click", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`
+      <div>
+        <div class="outside">outside</div>
+        <Dropdown>
+          <t t-set-slot="menu">
+            <Dropdown>
+              <t t-set-slot="menu">
+                <Dropdown/>
+              </t>
+            </Dropdown>
+          </t>
+        </Dropdown>
+      </div>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "div.outside");
```

Comment:
as a sanity check, assert above that 3 dropdowns are open

Reply:
:+1: